todo-list-backend/tests/integration/graphql-api.spec.ts

```diff
@@ -0,0 +1,526 @@
+                       mutation: UPDATE_TODO_DONE,
+                       variables: {id: todoFromOtherUser.id, done: true}
+                   });
+
+                   expect(response.data.updateTodoDone).toBeNull();
+               });
+
+               it("doesn't change the todo in storage", async () => {
+                   let response = await mutate({
+                       mutation: UPDATE_TODO_DONE,
+                       variables: {id: todoFromOtherUser.id, done: true}
+                   });
+
+                   expect(storage.readTodo(otherUser, todoFromOtherUser.id).done).toBeFalsy();
+               });
+           });
+
+           describe("deleteTodo", () => {
+               it("returns null", async () => {
+                   let response = await mutate({
+                       mutation: DELETE_TODO,
+                       variables: {
+                           id: todoFromOtherUser.id
+                       }
+                   });
+
+                   expect(response.data.deleteTodo).toBeNull();
+               });
+
+               it("doesn't remove todo from storage", async () => {
+                   let response = await mutate({
+                       mutation: DELETE_TODO,
+                       variables: {
+                           id: todoFromOtherUser.id
+                       }
+                   });
+
+                   expect(storage.readTodo(otherUser, todoFromOtherUser.id)).toBeTruthy();
+               });
+           });
+        });
+
+        describe("createUser", () => {
+            it("adds a new user to the storage", async () => {
+                let userName = "rudolphusTestus";
+
+                let response = await mutate({
+                    mutation: CREATE_USER,
+                    variables: {
+                        name: userName,
+                        password: "123"
+                    }
+                });
+
+                expect(storage.readUser(userName)).toMatchObject({name: userName});
+            });
+        });
+    });
+
+    describe("given a random authorization token", () => {
+        let storage: IStorage;
+        let query: any;
+        let mutate: any;
+
+        beforeEach(async () => {
+            storage = buildTwoUserInMemoryStorage(true);
+
+            const context = {token: "I'm not a valid token"};
+
+            const testServer = buildApolloServer(storage, authSecret, false, context);
+
+            const testClient = createTestClient(testServer);
+            query = testClient.query;
+            mutate = testClient.mutate;
+        });
+
+        describe("createTodo", () => {
+            it("raises an error", async () => {
+                let response = await mutate({
+                    mutation: CREATE_TODO,
+                    variables: {
+                        name: "new todo"
+                    }
+                });
+
+                expect(response.errors).toHaveLength(1);
```

Comment:
Here I would `expect(response).toMatchObject({ errors: [ { message: "Ouch" } ] })` the error message. It's helpful to understand your app and to specify the behaviour.

Reply:
Makes sense. We will change this